### PR TITLE
[Delivers #90162132] associate ApiTokens with Approvals

### PR DESCRIPTION
**Pull request by afeld**
Makes things easier for some forthcoming refactoring.

### CODE REVIEW

File: spec/lib/parallel_dispatcher_spec.rb [hunk 1]
```diff
@@ -13,9 +13,12 @@
     end
 
     it 'creates a new token for each approver' do
```
Comment (phirefly) [inline spec/lib/parallel_dispatcher_spec.rb]: Minor: The name of this specs seems a little off now

File: app/models/api_token.rb [hunk 2]
```diff
@@ -1,17 +1,26 @@
...
+  has_one :user, through: :approval
+
+  validates_presence_of :approval_id
 
   scope :unexpired, -> { where('expires_at >= ?', Time.now) }
   scope :expired, -> { where('expires_at < ?', Time.now) }
   scope :unused, -> { where(used_at: nil) }
   scope :fresh, -> { unused.unexpired }
 
+  def cart_id
```
Comment (phirefly) [inline app/models/api_token.rb]: We can delegate `cart_id` to `approval` like we do in [approval](https://github.com/18F/C2/blob/master/app/models/approval.rb#L10)

File: app/models/api_token.rb [hunk 3]
```diff
@@ -1,17 +1,26 @@
...
+    self.approval.cart_id
+  end
+
   def generate_token
```
Comment (phirefly) [inline app/models/api_token.rb]: I just noticed we don't have any specs for ApiToken. This method looks spec worthy.

File: app/models/approval.rb [hunk 4]
```diff
@@ -5,7 +5,9 @@ class Approval < ActiveRecord::Base
 
   belongs_to :cart
   belongs_to :user
+  has_one :api_token, -> { fresh }
```
Comment (phirefly) [inline app/models/approval.rb]: Neato

File: spec/lib/parallel_dispatcher_spec.rb (hunk 1 above)
Comment (afeld) [inline spec/lib/parallel_dispatcher_spec.rb]: I don't think it's inaccurate.

**Comment by afeld** [comment]
Updated!

**Comment by afeld** [comment]
Totally confused about why Timecop doesn't seem to be working in [the failing test](https://circleci.com/gh/18F/C2/578). Need to investigate.

**Comment by afeld** [comment]
Phew, fixed. Stupid floating point comparison! Anyway, ready for another look.

**Comment by cmc333333** [comment]
Splitting things up, one step at a time!

**Pull request merged**